### ISSUE TITLE
JSpecifyMode: Checking of :: reference to unmarked method is too strict

### ISSUE THREAD

**Issue by jeffrey-easyesi** [uber/NullAway#1195]
```java
Function<@Nullable Object, Boolean> myFilter = Objects::isNull;
// warning: [NullAway] parameter arg0 of referenced method is @NonNull, but parameter in functional interface method java.util.function.Function.apply(T) is @Nullable
```

This should be allowed, since `Objects.isNull(Object)` is unmarked (and does accept a nullable object), but it's being treated as if its parameter were `@NonNull` annotated.

**Workaround:** Using an explicit lambda+call, `o -> Objects.isNull(o)`, will work.
